src/test/java/com/puppycrawl/tools/checkstyle/ConfigurationLoaderTest.java

```diff
@@ -769,4 +769,20 @@ public void testLoadConfiguration3() throws Exception {
                     .contains("query");
         }
     }
+
+    @Test
+    public void testLengthInReplaceProperties() throws Exception {
+        final String[][] testValues = {
+                {"${a}"},
+                {"x${a}"},
```

Comment:
Why multi dimensional array ?

Reply:
converted into single dimension

Reply:
need to be converted to more full functional test based on xml files